File: insertionsort.js
const arr=[5,2,4,8,0 ]

const insertion=(arr)=>{
    for(let i=1;i<arr.length;i++){
        key=arr[i]
        j=i-1

        while(j>=0 && arr[j]>key){
            arr[j+1]=arr[j]
            j=j-1
        }
        arr[j+1]=key
    }
    return arr
}
console.log(arr);
insertion(arr)
console.log(arr);


// output:[ 5, 2, 4, 8, 0 ]
// [ 0, 2, 4, 5, 8 ]

File: Python/insertion sort.py

#................. insertion sort...................


def insertion(arr):
    
    for i in range(1,len(arr)):
        curr=arr[i]
        ind=i
        while arr[ind-1]>curr and ind>0:
            
            arr[ind]=arr[ind-1]
            ind-=1
        arr[ind]=curr
        
arr=[12,44,3,65,87,45]

insertion(arr)
print(arr)